### PR TITLE
Feature/anonymizer

### PR DESCRIPTION
**Pull request by davidt99**

### CODE REVIEW

File: intezer_analyze_cli/commands.py [hunk 1]
```diff
@@ -48,8 +48,12 @@ def analyze(self, send_options=None):
...
-        send_options = send_options or {}
-        core.send_code_item_for_analysis(self.code_item, self.api_proxy, **send_options)
+        if anonymize and not self.code_item.exists_in_cloud:
+            core.anonymize_file(self.code_item)
+
+        if not self.code_item.analysis_error:
+            send_options = send_options or {}
+            core.send_code_item_for_analysis(self.code_item, self.api_proxy, **send_options)
 
         if self.code_item.analysis_error:
```
Comment (shaytidhar) [inline intezer_analyze_cli/commands.py]: should be else

File: intezer_analyze_cli/core.py [hunk 2]
```diff
@@ -25,9 +27,10 @@
 
 
 class CodeItemMetadata(object):
-    __slots__ = ['sha256', 'code_item_type', 'file_path', 'analysis_id', 'exists_in_cloud', 'analysis_error']
+    __slots__ = ['sha256', 'code_item_type', 'file_path', 'analysis_id', 'exists_in_cloud', 'analysis_error',
```
Comment (shaytidhar) [inline intezer_analyze_cli/core.py]: lines

File: intezer_analyze_cli/core.py [hunk 3]
```diff
@@ -241,7 +292,8 @@ def send_code_item_for_analysis(code_item, api_proxy, **kwargs):
 
 
 def print_contact_us_if_needed(errors):
-    if any(x in errors for x in (AnalysisErrorType.GENERAL_ERROR, AnalysisErrorType.ANALYSIS_CREATION_ERROR)):
+    if any(x in errors for x in (AnalysisErrorType.GENERAL_ERROR, AnalysisErrorType.ANALYSIS_CREATION_ERROR,
```
Comment (shaytidhar) [inline intezer_analyze_cli/core.py]: lines

File: intezer_analyze_cli/core.py [hunk 4]
```diff
@@ -162,14 +166,58 @@ def check_code_items_exist_on_cloud(code_items, api_proxy):
...
+        code_item.analysis_error = AnalysisErrorType.ANONYMIZE_ERROR
+        return
+
+
+def anonymize_files(code_items):
+    """
+    Anonymize list of files using anonymizer executable
+    :param list[CodeItemMetadata] code_items: Code items list
+    """
+    with click.progressbar(length=len(code_items), label='Anonymizing files, it might take a while',
```
Comment (shaytidhar) [inline intezer_analyze_cli/core.py]: shouldnt be also on data types

File: intezer_analyze_cli/commands.py (hunk 1 above)
Comment (davidt99) [reply intezer_analyze_cli/commands.py]: `core.send_code_item_for_analysis` can change the code item's error

File: intezer_analyze_cli/core.py (hunk 2 above)
Comment (davidt99) [reply intezer_analyze_cli/core.py]: Done

File: intezer_analyze_cli/core.py (hunk 3 above)
Comment (davidt99) [reply intezer_analyze_cli/core.py]: Done

File: intezer_analyze_cli/core.py (hunk 4 above)
Comment (davidt99) [reply intezer_analyze_cli/core.py]: We talked

**Pull request merged**